homeassistant/components/shelly/utils.py

```diff
@@ -346,3 +346,30 @@ def get_rpc_input_triggers(device: RpcDevice) -> list[tuple[str, str]]:
             triggers.append((trigger_type, subtype))
 
     return triggers
+
+
+async def async_device_update_info(
+    hass: HomeAssistant, shellydevice: BlockDevice | RpcDevice, shellyentry: ConfigEntry
+) -> None:
+    """Update device registry info."""
+
+    _LOGGER.debug("Updating device registry info")
+    dev_registry = device_registry.async_get(hass)
+    device = None
+    if shellyentry.unique_id:
+        device = dev_registry.async_get_device(
+            identifiers={(DOMAIN, shellyentry.entry_id)},
+            connections={
+                (
+                    device_registry.CONNECTION_NETWORK_MAC,
+                    device_registry.format_mac(shellyentry.unique_id),
+                )
+            },
+        )
+    if device is not None:
+        dev_registry.async_update_device(
+            device.id,
+            sw_version=shellydevice.firmware_version
+            if shellydevice.initialized
+            else "",
```

Comment:
This does not make any sense to reach the update method and update the firmware version to "", if we reach here it is because we have a new firmware version value

Reply:
Indeed, was just a need for making mypy happy ;-)

Reply:
Actually not, code updated